Converts WFJTForm to Formik hooks

AlexSCorey opened the pull request:
##### SUMMARY
This addresses #6384.  It add formik hooks to wfjt form.  There is still a lot of room for improvement in wfjtform.jsx file and I have create an issue for it https://github.com/ansible/awx/issues/6546


##### ISSUE TYPE
-enhancement

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
9.3
```


##### ADDITIONAL INFORMATION

AlexSCorey committed:
Adds formik hook functionality to wfjt form

AlexSCorey committed:
Adds tests

AlexSCorey commented:
recheck

jlmitch5 commented on awx/ui_next/src/screens/Template/WorkflowJobTemplateAdd/WorkflowJobTemplateAdd.jsx:
```diff
@@ -12,7 +12,19 @@ function WorkflowJobTemplateAdd() {
+      organization,
+      webhook_credential,
+      webhookKey,
+      ...remainingValues
```
this would be better named postPayload or something...it was confusing to me that you were adding things to "remainingValues" below on first glance.

jlmitch5 commented on awx/ui_next/src/screens/Template/WorkflowJobTemplateAdd/WorkflowJobTemplateAdd.jsx:
```diff
@@ -12,7 +12,19 @@ function WorkflowJobTemplateAdd() {
+    remainingValues.organization = organization?.id;
+    remainingValues.webhook_credential = webhook_credential?.id;
+    const organizationId =
+      organization?.id || inventory?.summary_fields?.organization.id || null;
```
you don't need || null here.

jlmitch5 commented on awx/ui_next/src/screens/Template/WorkflowJobTemplateAdd/WorkflowJobTemplateAdd.test.jsx:
```diff
@@ -63,16 +70,49 @@ describe('<WorkflowJobTemplateAdd/>', () => {
+      wrapper
+        .find('SelectOption')
+        .find('button')
+        .at(2)
```
Is there a way to make this select more definitive than "at(2)", maybe using button[aria-label="Foo"]?

jlmitch5 commented on awx/ui_next/src/screens/Template/WorkflowJobTemplateEdit/WorkflowJobTemplateEdit.test.jsx:
```diff
@@ -71,29 +76,67 @@ describe('<WorkflowJobTemplateEdit/>', () => {
+      wrapper
+        .find('SelectOption')
+        .find('button')
+        .at(0)
```
same

jlmitch5 commented on awx/ui_next/src/screens/Template/WorkflowJobTemplateEdit/WorkflowJobTemplateEdit.test.jsx:
```diff
@@ -71,29 +76,67 @@ describe('<WorkflowJobTemplateEdit/>', () => {
+      wrapper
+        .find('SelectOption')
+        .find('button')
+        .at(2)
```
same

jlmitch5 commented on awx/ui_next/src/screens/Template/shared/WorkflowJobTemplateForm.jsx:
```diff
@@ -1,11 +1,11 @@
 import React, { useState, useEffect, useCallback } from 'react';
 import { t } from '@lingui/macro';
-import { useRouteMatch, useParams } from 'react-router-dom';
+import { useRouteMatch, useParams, withRouter } from 'react-router-dom';
```
I don't think you need withRouter?

jlmitch5 commented on awx/ui_next/src/screens/Template/WorkflowJobTemplateEdit/WorkflowJobTemplateEdit.jsx:
```diff
@@ -60,7 +73,7 @@ function WorkflowJobTemplateEdit({ template, webhook_key }) {
         handleCancel={handleCancel}
         template={template}
-        webhook_key={webhook_key}
+        webhookKey={webhook_key}
```
maybe put this on template so it's like there's one object prop that has all the data the form needs to render.

jlmitch5 reviewed (commented):
awesome stuff!

AlexSCorey commented:
recheck

AlexSCorey committed:
Puts webhook key on the template object in WFJTEdit
 Also adds aria-label to Label Select Options to improve test matchers
 Improves the name of the template payload in WFJTAdd and WFJTEdit
 Updates tests including a failing snapshot DeleteConfirmationModal
 test that was failing in devel

AlexSCorey commented:
regate

AlexSCorey commented:
regate

jakemcdermott commented:
regate

AlexSCorey commented:
regate
